admin_ui/src/components/RowForm.vue

```diff
@@ -75,68 +72,35 @@ export default Vue.extend({
         schema: Object,
         isFilter: {
             type: Boolean,
-            default: false
-        }
+            default: false,
+        },
     },
     components: {
         InputField,
-        KeySelect
+        KeySelect,
     },
     data() {
         return {
             keySelectIDs: {},
-            baseIndex: 1,
-            windowListener: undefined
         }
     },
     methods: {
         getValue(propertyTitle: string) {
             let value = this.row
-                ? this.row[
-                      propertyTitle
-                          .toLowerCase()
-                          .split(" ")
-                          .join("_")
-                  ]
+                ? this.row[propertyTitle.toLowerCase().split(" ").join("_")]
                 : undefined
             return value
         },
         getKeySelectID(propertyTitle: string) {
-            return (
-                this.keySelectIDs[propertyTitle] || this.getValue(propertyTitle)
-            )
-        },
-        keySelectChanged(propertyTitle: string, value: number) {
-            console.log(`${propertyTitle} = ${value}`)
-            Vue.set(this.keySelectIDs, propertyTitle, value)
-        },
-        // We use this to refresh KeySelect components
-        getKey(keyName: string) {
-            return keyName + this.baseIndex
+            return this.getValue(propertyTitle)
         },
         isRequired(keyName: string) {
             return (
                 !this.isFilter &&
                 (this.schema.required || []).indexOf(keyName) != -1
             )
-        }
-    },
-    mounted() {
-        window.addEventListener("message", receiveMessage, false)
-
-        let app = this
-
-        function receiveMessage(event) {
-            if (event.origin !== document.location.origin) return
-            console.log("Received message")
-            app.baseIndex = app.baseIndex + 1
-        }
+        },
     },
-    destroyed() {
-        if (this.windowListener) {
-            window.removeEventListener("message", this.windowListener)
-        }
-    }
```

Comment:
This stuff seems quite cryptic, but when you click on the edit or add button above the dropdown, it opens a new tab, and this code will tell the `KeySelect` element to refresh (and hence pull down fresh data from the API, containing the new / modified rows).

You'll see the bit of code which `opener.postMessage("edited row", document.location.origin)` in `EditRowForm`.

Reply:
@dantownsend Thanks for the insight and explanation, it's clear now. I deleted this code because the component was refreshing and I had to select the value from the search dropdown twice.

Reply:
It's a very obscure API which lets different tabs communicate with each other.

It does create weird issues though - sometimes when Chrome dev tools is open it sends loads of messages for some reason.